src/react/workflow/ReplicationForm.tsx

```diff
@@ -386,104 +410,181 @@ const RenderDestination = ({
   locations: Locations;
   errors: Record<string, FieldError>;
   touchedFields: { [key: string]: boolean };
+  isTransient: boolean;
 }) => {
   const forceLabelWidth = convertRemToPixels(10);
   const { trigger } = useFormContext();
+  const options = destinationOptions(locations);
   return (
-    <Controller
-      control={control}
-      name={name}
-      render={({
-        field: { onChange, onBlur, value: destinationLocations },
-      }) => (
-        <FormSection
-          title={{ name: 'Destination' }}
-          forceLabelWidth={forceLabelWidth}
-        >
-          {!Array.isArray(destinationLocations)
-            ? []
-            : destinationLocations.map((destLoc: string, index) => {
-                const options = destinationOptions(locations);
-                const fieldName = `${prefix}destinationLocation.${index}`;
-                const err = errors[fieldName];
-                const touched = touchedFields[fieldName];
-                return (
-                  <FormGroup
-                    required={index === 0}
-                    label={index === 0 ? 'Location Name' : ''}
-                    id={`select-location-${index}`}
-                    error={touched ? err?.message : undefined}
-                    helpErrorPosition="right"
-                    content={
-                      <Box
-                        gap={spacing.r8}
-                        display="flex"
-                        justifyContent="space-between"
-                        alignItems="center"
-                        data-testid={`select-location-name-replication-${index}`}
-                      >
-                        <Select
-                          onBlur={onBlur}
-                          id="destinationLocation"
-                          onChange={(value) => {
-                            const newValues = [...destinationLocations];
-                            newValues[index] = value;
-                            onChange(newValues);
-                            trigger();
-                          }}
-                          value={destLoc}
+    <FormSection
+      title={{ name: 'Destination' }}
+      forceLabelWidth={forceLabelWidth}
+    >
```

Comment:
The code and logic look very good to me! 

Just the `Destination` Section is not optional. 
We indeed already have the logic within <Form/> component in core-ui to compute whether it's optional. 
I guess the logic doesn't meet this specific case, I'll let you to check if there is a way to quickly fix this. 

And by the way maybe we can increase a bit the `forceLabelWidth`, in order to put `Preferred Read Location *` in one line?

Thanks! 
![image](https://user-images.githubusercontent.com/18453133/200849904-20fdd713-a0ee-49ff-85a0-1ff045470b33.png)